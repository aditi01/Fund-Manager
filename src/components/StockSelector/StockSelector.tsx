import { Select } from 'antd';

const { Option } = Select;

interface IProps {
  stocks: string[],
  selectedStocks: string[],
  onChange: (stocks: string[]) => void
}

const StockSelector = (props: IProps) => {
  const { stocks, selectedStocks, onChange } = props
  const handleChange = (e: any) => {
    const value = e.slice();
    if (value.length <= 3) {
      onChange(value);
    }
  };

  return (
    <Select
      data-testid="select-stock"
      mode="multiple"
      defaultValue={selectedStocks}
      style={{ width: '100%' }}
      placeholder="Select stocks"
      onChange={handleChange}
    >
      {stocks.map(stock => (
        <Option data-testid="select-option" key={stock} value={stock} disabled={selectedStocks.length >= 3 && !selectedStocks.includes(stock)}>
          {stock}
        </Option>
      ))}
    </Select>
  );
};

export default StockSelector;
