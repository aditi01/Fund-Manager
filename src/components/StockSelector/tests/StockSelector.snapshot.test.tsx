// StockSelector.test.tsx
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import StockSelector from '../StockSelector';

describe('StockSelector', () => {
  const stocks = ['AAPL', 'MSFT', 'AMZN', 'FB', 'GOOGL', 'TSLA'];
  const selectedStocks = ['AAPL', 'MSFT'];
  const mockOnChange = jest.fn();

  it('renders correctly and matches snapshot', () => {
    const { container } = render(
      <StockSelector stocks={stocks} selectedStocks={selectedStocks} onChange={mockOnChange} />
    );

    expect(container).toMatchSnapshot();
  });
});
