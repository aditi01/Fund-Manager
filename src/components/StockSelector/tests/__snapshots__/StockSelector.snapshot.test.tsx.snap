// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockSelector renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="ant-select ant-select-outlined css-dev-only-do-not-override-3rel02 ant-select-multiple ant-select-show-arrow ant-select-show-search"
    data-testid="select-stock"
    style="width: 100%;"
  >
    <div
      class="ant-select-selector"
    >
      <div
        class="ant-select-selection-overflow"
      >
        <div
          class="ant-select-selection-overflow-item"
          style="opacity: 1;"
        >
          <span
            class="ant-select-selection-item"
            title="AAPL"
          >
            <span
              class="ant-select-selection-item-content"
            >
              AAPL
            </span>
            <span
              aria-hidden="true"
              class="ant-select-selection-item-remove"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="close"
                class="anticon anticon-close"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <div
          class="ant-select-selection-overflow-item"
          style="opacity: 1;"
        >
          <span
            class="ant-select-selection-item"
            title="MSFT"
          >
            <span
              class="ant-select-selection-item-content"
            >
              MSFT
            </span>
            <span
              aria-hidden="true"
              class="ant-select-selection-item-remove"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="close"
                class="anticon anticon-close"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close"
                  fill="currentColor"
                  fill-rule="evenodd"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <div
          class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
          style="opacity: 1;"
        >
          <div
            class="ant-select-selection-search"
            style="width: 0px;"
          >
            <input
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              readonly=""
              role="combobox"
              style="opacity: 0;"
              type="search"
              unselectable="on"
              value=""
            />
            <span
              aria-hidden="true"
              class="ant-select-selection-search-mirror"
            >
              Â 
            </span>
          </div>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      class="ant-select-arrow"
      style="user-select: none;"
      unselectable="on"
    >
      <span
        aria-label="down"
        class="anticon anticon-down ant-select-suffix"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </span>
    </span>
  </div>
</div>
`;
